### PR TITLE
Homepage with added functionalities

### PR DESCRIPTION
**Pull request by Shailseen**
### Home Page
**Features:**
`Login`: using context for login-authentication

`Active link`: Selected menu  show as active page, which one is selected.

`Rich-Text-Feature-Card`: User have access to use rich-text-feature for styling his notes & card created with detail like `Created date`.

`Edit card`: Edit feature is not added but UI is added for it.

`Routes`: using routes for automatically user to homepage if login successfully.

`Toast`: using toast for when user login or card added successfully.

  
Please review and give your feedback :)

https://user-images.githubusercontent.com/90495133/161636346-7172cc44-589c-4a9f-a701-67ddb3701d89.mp4

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+      <Toast/>
+      {/* <BrowserRouter> */}
         <Routes>
           <Route path="/" element={<LandingPage />} />
+          <Route path="/home" element={<HomePage/>} />
+          <Route path="/login" element={<LoginPage/>}/>
+          <Route path="/mock" element={<Mockman />} />
         </Routes>
-      </BrowserRouter>
+      {/* </BrowserRouter> */}
```
Comment (hrishikeshmane) [inline src/App.js]: remove commented code

File: src/App.js [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+import { LoginPage } from "./routes/loginpage/loginPage";
+import { Toast } from "./toast/toast";
+import Mockman from "mockman-js";
+import "../node_modules/mockman-js/dist/style.css";
 function App() {
   return (
     <div className="App">
-      <BrowserRouter>
+      <Toast/>
+      {/* <BrowserRouter> */}
```
Comment (hrishikeshmane) [inline src/App.js]: remove commented code

File: src/component/notespage/notes-page.css [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.notes-display-container{
+    grid-area: "notes-display";
+    /* border: 2px solid yellowgreen; */
```
Comment (hrishikeshmane) [inline src/component/notespage/notes-page.css]: this too

File: src/component/notespage/notes-page.jsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import "./notes-page.css"
+
+export const NotesPage = () => {
+    const {notesList} = useNotes()
+    return (
+        <section className="notes-display-container">
+        <SearchBar/>
+        <AddCard/>
+        {notesList.length !== 0 && notesList.map((item) => <NotesCard key={item._id} item={item}/>)}
+        </section>
```
Comment (hrishikeshmane) [inline src/component/notespage/notes-page.jsx]: format code with prettier

File: src/component/searchbar/searchbar.jsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import "./searchbar.css"
+import SearchOutlinedIcon from '@mui/icons-material/SearchOutlined'
+import TuneOutlinedIcon from '@mui/icons-material/TuneOutlined';
+export const SearchBar = () => {
+    return (
+        <div className="searchbar-container">
+        <SearchOutlinedIcon className="searchbar-icon"/>
+        <input className="searchbar-text" type="text" placeholder="Search"/>
+        <TuneOutlinedIcon className="filter-button"/>
+        </div>
```
Comment (hrishikeshmane) [inline src/component/searchbar/searchbar.jsx:10]: here aswell.

**Review by hrishikeshmane — APPROVED** [review]
Looks good just remove commented code and format before pushing.

File: src/App.js (hunk 1 above)
Comment (Shailseen) [reply src/App.js]: okay .

File: src/App.js (hunk 2 above)
Comment (Shailseen) [reply src/App.js]: sure.

File: src/component/add-card/addCard.jsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+          className="round-colors color-violet"
+        ></div>
+        <div
+          onClick={() => containerBackgroundColorHandler("color-grey")}
+          className="round-colors color-grey"
+        ></div>
+        <div
+          onClick={() => containerBackgroundColorHandler("color-pink")}
+          className="round-colors color-pink"
+        ></div>
```
Comment (Aditi0505) [inline src/component/add-card/addCard.jsx]: You can create an array of colors and map for it to reduce no of lines of code.

File: src/component/notesCard/notes-card.jsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        <div
+          onClick={() => containerBackgroundColorHandler("color-violet")}
+          className="round-colors color-violet"
+        ></div>
+        <div
+          onClick={() => containerBackgroundColorHandler("color-grey")}
+          className="round-colors color-grey"
+        ></div>
+        <div
+          onClick={() => containerBackgroundColorHandler("color-pink")}
```
Comment (Aditi0505) [inline src/component/notesCard/notes-card.jsx]: Same as above

File: src/context/notes-context.js [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+        isDisplay: "visible",
+      }));
+      setTimeout(() => {
+        setToastVal((prevVal) => ({
+          ...prevVal,
+          isDisplay: "hidden",
+        }));
+      }, 2000);
+    } catch (error) {
+      console.log(error);
```
Comment (Aditi0505) [inline src/context/notes-context.js]: Use toast here as well

File: src/routes/loginpage/loginPage.css [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+}
+.password-container {
+  display: flex;
+  width: 100%;
+}
+
+.show-hide-pass {
+  padding: 0rem;
+  padding-right: 0.5rem;
+  /* height: 1.4rem; */
```
Comment (Aditi0505) [inline src/routes/loginpage/loginPage.css]: Please remove commented code.

**Review by Aditi0505 — APPROVED** [review]
Left few suggestions.

- Clean code 
- Good folder structure followed
- Good handle of error states using toast

File: src/component/add-card/addCard.jsx (hunk 6 above)
Comment (Shailseen) [reply src/component/add-card/addCard.jsx]: thanks for giving suggestion i will do changes.

File: src/component/notesCard/notes-card.jsx (hunk 7 above)
Comment (Shailseen) [reply src/component/notesCard/notes-card.jsx]: okay.

File: src/context/notes-context.js (hunk 8 above)
Comment (Shailseen) [reply src/context/notes-context.js]: sure.

File: src/routes/loginpage/loginPage.css (hunk 9 above)
Comment (Shailseen) [reply src/routes/loginpage/loginPage.css]: okay.

**Pull request merged**